spacepy/toolbox/__init__.py

```diff
@@ -465,6 +468,7 @@ def savepickle(fln, dict, compress=None):
     >>> savepickle('test.pbin', d)
     """
     if compress == None:
+        # TODO, what is this line meant to do? Whay not just check fln for ending in gz?
```

Comment:
I believe that code is unreachable or misleading at best. The file is over written anyway so what does it matter if it existed with a gz...

Reply:
No, it's not unreachable. If you call `savepickle('foo', data, compress=False)` it writes to `foo`. If you call `savepickle('foo', data, compress=True)` it writes to `foo.gz`. if you call `savepickle('foo', data)` it will write to `foo` unless `foo` does not exist and `foo.gz` does, in which case it writes to `foo.gz`. As the docstring says, "write as a gzip-compressed file (.gz will be added to L{fln}). If not specified, defaults to uncompressed, unless the compressed file exists and the uncompressed does not." Ignore the epydoc, please.

We don't have an explicit _test_ of the case where the compressed version exists and `compress` is not specified.